rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalTransferServer.java

```diff
@@ -0,0 +1,191 @@
+                    threadPool.execute(cliThread);
+                }
+            }
+            catch (AsynchronousCloseException e) {
+                log.info("Stopping Local Transfer server");
+            }
+            catch (IOException e) {
+                log.error(String.format("Error starting Local Transfer server %s", Throwables.getStackTraceAsString(e)));
+                e.printStackTrace();
+            }
+
+        }
+
+        public void stop()
+        {
+            try {
+                listener.close();
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+    static class ClientServiceThread
+            extends Thread
+    {
+        SocketChannel localTransferClient;
+        RetryingBookkeeperClient bookKeeperClient;
+
+        ClientServiceThread(SocketChannel s)
+        {
+            localTransferClient = s;
+        }
+
+        public void run()
+        {
+            try {
+                log.debug("Connected to node - " + localTransferClient.getLocalAddress());
+                BookKeeperFactory bookKeeperFactory = new BookKeeperFactory();
+                ByteBuffer dataInfo = ByteBuffer.allocate(CacheConfig.getDataTransferBufferSize(conf));
+
+                int read = localTransferClient.read(dataInfo);
+                if (read == -1) {
+                    localTransferClient.close();
+                    return;
+                }
+                dataInfo.flip();
+                /* order is: long : offset, int : readLength, long : fileSize, long : lastModified,
+                int : clusterType, int : filePathLength, String : filePath */
+                long offset = dataInfo.getLong();
+                int readLength = dataInfo.getInt();
+                long fileSize = dataInfo.getLong();
+                long lastModified = dataInfo.getLong();
+                int clusterType = dataInfo.getInt();
+                int filePathLength = dataInfo.getInt();
+                byte[] fileBytes = new byte[filePathLength];
+                dataInfo.get(fileBytes);
+                String remotePath = new String(fileBytes);
+                try {
+                    bookKeeperClient = bookKeeperFactory.createBookKeeperClient(conf);
+                    if (!bookKeeperClient.readData(remotePath, offset, readLength, fileSize, lastModified, clusterType)) {
+                        localTransferClient.close();
+                    }
+                }
+                catch (TException e) {
+                    log.info("Could not create BookKeeper client" + Throwables.getStackTraceAsString(e));
+                    localTransferClient.close();
+                }
+
+                String filename = CacheConfig.getLocalPath(remotePath, conf);
+                FileChannel fc = new FileInputStream(filename).getChannel();
+                int maxCount = (10 * 1024 * 1024);
```

Comment:
keep this configurable too? I know 10MB was the best performing number but never know that under different circumstances what value would be better, keeping it configurable will also help us to experiment easily